private/scribble-util.rkt

```diff
@@ -0,0 +1,24 @@
+#lang racket/base
+
+(module doc racket/base
+
+  (require racket/contract/base
+           scribble/base
+           scribble/core
+           scribble/decode
+           scribble/manual)
+
+  (provide
+    (contract-out
+      [tech/guide
+        ;; Similar to @tech{....}, but creates a link to a `deftech` in The Racket Guide
+        (->* () (#:key (or/c string? #f)) #:rest (listof pre-content?) element?)]
+      [tech/reference
+        ;; Links to a `deftech` in The Racket Reference
+        (->* () (#:key (or/c string? #f)) #:rest (listof pre-content?) element?)]))
+
+  (define (tech/guide #:key [key #f] . text)
+    (keyword-apply tech '(#:doc #:key) `((lib "scribblings/guide/guide.scrbl") ,key) text))
```

Comment:
Can we avoid / minimize the quoting here? Possibly by doing something like this:

```racket
(define guide-module '(lib "scribblings/guide/guide.scrbl"))
(define reference-module '(lib "scribblings/reference/reference.scrbl"))

(define (tech/guide ...)
  (keyword-apply tech (list '#:doc '#:key) (list guide-module key) text)

(define (tech/reference ...) ...)
```

I try to limit quoting to only atomic data like `'foo` and `'#:bar`, with the exception of module paths. StackOverflow is littered with the remains of students defeated by quoted lists.